lib/streamlit/testing/element_tree.py

```diff
@@ -958,6 +967,130 @@ def decrement(self) -> NumberInput:
+
+    root: ElementTree = field(repr=False)
+
+    def __init__(self, proto: DateInputProto, root: ElementTree):
+        self.proto = proto
+        self.root = root
+        self._value = None
+
+        self.type = "date_input"
+        self.id = proto.id
+        self.label = proto.label
+        self.min = datetime.strptime(proto.min, "%Y/%m/%d").date()
+        self.max = datetime.strptime(proto.max, "%Y/%m/%d").date()
+        self.is_range = proto.is_range
+        self.help = proto.help
+        self.form_id = proto.form_id
+        self.disabled = proto.disabled
+        self.key = user_key_from_widget_id(self.id)
+
+    def set_value(self, v: DateValue) -> DateInput:
+        self._value = v
+        return self
+
+    def widget_state(self) -> WidgetState:
+        ws = WidgetState()
+        ws.id = self.id
+
+        serde = DateInputSerde(None)  # type: ignore
+        ws.string_array_value.data[:] = serde.serialize(self.value)
+        return ws
+
+    @property
+    def value(self) -> DateWidgetReturn:
+        if self._value is not None:
+            parsed, _ = _parse_date_value(self._value)
+            return tuple(parsed)  # type: ignore
+        else:
+            state = self.root.session_state
+            assert state
+            return state[self.id]  # type: ignore
+
+
+TimeValue: TypeAlias = Union[time, datetime]
+
+
+@dataclass(repr=False)
+class TimeInput(Element, Widget):
+    _value: TimeValue | None
+    proto: TimeInputProto
+    type: str
+    id: str
+    label: str
+    step: int
+    help: str
+    form_id: str
+    disabled: bool
+    key: str | None
+
+    root: ElementTree = field(repr=False)
+
+    def __init__(self, proto: TimeInputProto, root: ElementTree):
+        self.proto = proto
+        self.root = root
+        self._value = None
+
+        self.type = "time_input"
+        self.id = proto.id
+        self.label = proto.label
+        self.step = proto.step
+        self.help = proto.help
+        self.form_id = proto.form_id
+        self.disabled = proto.disabled
+        self.key = user_key_from_widget_id(self.id)
+
+    def set_value(self, v: TimeValue) -> TimeInput:
+        self._value = v
+        return self
+
+    def widget_state(self) -> WidgetState:
+        ws = WidgetState()
+        ws.id = self.id
+
+        serde = TimeInputSerde(None)  # type: ignore
+        ws.string_value = serde.serialize(self.value)
+        return ws
+
+    @property
+    def value(self) -> time:
+        if self._value is not None:
+            v = self._value
+            v = v.time() if isinstance(v, datetime) else v
+            return v
+        else:
+            state = self.root.session_state
+            assert state
+            return state[self.id]  # type: ignore
+
+    def increment(self) -> TimeInput:
+        dt = datetime.combine(date.today(), self.value) + timedelta(seconds=self.step)
+        return self.set_value(dt.time())
+
+    def decrement(self) -> TimeInput:
```

Comment:
Can you add docstrings for these? How much do they increment by, and also - do they need to be synchronized with any frontend code? (Like, do we need to make sure that TimeInput.increment here matches the behavior of TimeInput.increment on the frontend, or is that not important?)